### PR TITLE
Added ROCK kernel which is required by AMDs ROCm project

### PR DESCRIPTION
**Pull request by alexanderkjeldaas**
Make the ROCm OpenCL and ecosystem work.

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/os-specific/linux/kernel/linux-4.11-kfd.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import ./generic.nix (args // rec {
+  version = "${ver}-${revision}";
+  modDirVersion = "${ver}";
+  extraMeta.branch = "4.11";
+
+  src = fetchFromGitHub {
+    owner = "RadeonOpenCompute";
+    repo = "ROCK-Kernel-Driver";
+    # url = "https://github.com/RadeonOpenCompute/ROCK-Kernel-Driver.git";
+    rev = "roc-1.6.x";
```
Comment (Mic92) [inline pkgs/os-specific/linux/kernel/linux-4.11-kfd.nix]: This is a branch, can you use a `tag` instead? Otherwise the source hash will break.

Comment (joachifm) [reply pkgs/os-specific/linux/kernel/linux-4.11-kfd.nix]: The latest tag is for 1.5 it seems, perhaps that is too old? In that case, using the commit id might be a little better than just the branch, until a tag is created.

**Review by matthewbauer — CHANGES_REQUESTED** [review]
See @Mic92's comment

**Comment by c0bw3b** [comment]
This is outdated now.
@alexanderkjeldaas feel free to come up with a new PR if you still want to bring ROCK kernels to nixpkgs.
There is releases based on kernel 4.14 and 4.15 now it seems.

**Pull request closed without merging**